[Lucas Johnson] iP

eman-kom opened the pull request:
# This is a Duke Bot

>Do today, not ~tomorrow~ 👈 😤

- Create **Todo** Tasks
- Create **Event** Tasks
- Create **Deadline** Tasks

1. You can *find* tasks
2. You can *save* tasks
3. You can *delete* tasks

- [x] Mark tasks as completed
- [ ] Mark tasks as not completed

```java
System.out.println("System.out.println.");
```

Use `git status` to get the status of your git.


[Link to the docs page](https://eman-kom.github.io/ip/).

Commit:
Add Gradle support

eman-kom committed:
Greet, Echo, Exit

eman-kom committed:
Add, List

eman-kom committed:
Created Task Class

eman-kom committed:
Mark as Done

eman-kom committed:
Created Deadline, Event, ToDo subclasses

eman-kom committed:
ToDo, Event, Deadline

eman-kom committed:
Automated Text UI Testing

eman-kom committed:
Handle Errors

eman-kom committed:
Delete

eman-kom committed:
Refactored some code

eman-kom committed:
Save

eman-kom committed:
Implement Dates and Times

eman-kom committed:
Merge branch 'branch-Level-7'

eman-kom committed:
Merge branch 'branch-Level-8'

eman-kom committed:
Use More OOP

eman-kom committed:
Organize into Packages

eman-kom committed:
Add JUnit Tests

eman-kom committed:
Implement Find function

eman-kom committed:
Follow the Coding Standard

eman-kom committed:
Add Javadocs

eman-kom committed:
Merge branch 'branch-A-JavaDoc'

eman-kom committed:
merge with branch-A-CodingStandard

eman-kom committed:
merge with branch-Level-9

eman-kom committed:
Repackage some modules further

eman-kom committed:
Merge remote-tracking branch 'origin/add-gradle-support'

eman-kom committed:
Initialise Gradle

eman-kom committed:
Initialise a checkstyle xml file to be used in the project

Wei-Xinn commented on src/main/java/seedu/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+public class Storage {
+
+    private final File file;
+    private final SimpleDateFormat ft = new SimpleDateFormat("dd MMM yyy h.mma");
```
```suggestion
    private final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("dd MMM yyy h.mma");
```
Consider changing naming variable to match naming convention for constants.

Wei-Xinn commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,189 @@
+        String[] parsedTask;
+        switch (cmd.toLowerCase().trim()) {
+
+        case "list":
+            System.out.println(tasks.toString());
+            break;
+
+        case "mark":
```
```suggestion
        case "list":
            System.out.println(tasks.toString());
            break;
        case "mark":
```
I am not very sure about this, but the coding standard should be that there shouldn't be a line break after "break;"

Wei-Xinn commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+        boolean isQuit = false;
+
+        while (!isQuit) {
+            try {
+
+                String input = ui.getCommand();
+                String[] cmd = parser.splitCmd(input);
+
+                if (parser.checkQuit(cmd[0].trim())) {
+                    isQuit = true;
+                } else {
+                    String task = cmd.length < 2 ? "" : cmd[1];
+                    parser.runCmd(cmd[0].trim(), task.trim(), tasks, ui, storage);
+                }
+
+            } catch (DukeException | IOException e) {
+                ui.showError(e.toString());
+            }
```
```suggestion
            try {
                String input = ui.getCommand();
                String[] cmd = parser.splitCmd(input);

                if (parser.checkQuit(cmd[0].trim())) {
                    isQuit = true;
                } else {
                    String task = cmd.length < 2 ? "" : cmd[1];
                    parser.runCmd(cmd[0].trim(), task.trim(), tasks, ui, storage);
                }
            } catch (DukeException | IOException e) {
                ui.showError(e.toString());
            }
```
Good line breaks to make it the logic clear and readable!
But I do think that there is a need for the line breaks after try and before catch

Wei-Xinn commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,189 @@
+
+public class Parser {
+
+    private final String[] cmdTypes = {"find", "list", "delete", "unmark", "mark", "todo", "event", "deadline", "bye"};
```
```suggestion
    private final String[] CMD_TYPES = {"find", "list", "delete", "unmark", "mark", "todo", "event", "deadline", "bye"};
```
Consider changing variable name to match constant naming convention.

Wei-Xinn commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,189 @@
+
+        String[] splitTask = task.split(keyWord, 2);
+
+        if (splitTask.length < 2) {
+            throw new DukeException("\"" + keyWord + "\" Not Found");
+        } else {
+            if (splitTask[0].equals("") && splitTask[1].equals("")) {
+                throw new DukeException("No task in instruction");
+            } else if (splitTask[0].equals("")) {
+                throw new DukeException("Description of task not found");
+            } else if (splitTask[1].equals("")) {
+                throw new DukeException("Datetime of task not found");
+            }
+        }
```
Concise and readable!

Wei-Xinn commented on src/main/java/seedu/task/Event.java:
```diff
@@ -0,0 +1,32 @@
+public class Event extends Task {
+
+    private final Date duration;
+    private final SimpleDateFormat ft = new SimpleDateFormat("dd MMM yyy h.mma");
```
Same as above. Maybe consider changing variable name to match constant naming convention

Wei-Xinn commented on src/main/java/seedu/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+public class Task {
+
+    private final String description;
+    private boolean completed;
```
```suggestion
    private boolean isCompleted;
```
Would isCompleted be a more intuitive vairable name?

hanqinilnix commented on src/main/java/seedu/duke/Ui.java:
```diff
@@ -0,0 +1,72 @@
+
+public class Ui {
+
+    private final Scanner sc;
```
Constant variables should be capitalised according to the coding standards. I noticed similar issues in other areas as well.

hanqinilnix commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+            try {
+
+                String input = ui.getCommand();
+                String[] cmd = parser.splitCmd(input);
+
+                if (parser.checkQuit(cmd[0].trim())) {
+                    isQuit = true;
+                } else {
+                    String task = cmd.length < 2 ? "" : cmd[1];
+                    parser.runCmd(cmd[0].trim(), task.trim(), tasks, ui, storage);
+                }
```
Wouldn't it be better if you encapsulated this in the Parser class?

hanqinilnix commented on src/main/java/seedu/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        while (sc.hasNext()) {
+            String[] task = sc.nextLine().trim().split("\t");
```
```suggestion
            String[] taskSegments = sc.nextLine().trim().split("\t");
```
Perhaps you can nouns that reflect the variable better. Plural form is also be used when representing a collection of objects.

hanqinilnix commented on src/main/java/seedu/task/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    /**
+     * @inheritDoc
```
Interesting annotation

hanqinilnix commented on src/main/java/seedu/task/Task.java:
```diff
@@ -0,0 +1,58 @@
+     * @param description of task
+     * @param completed task is completed or not
+     */
+    public Task(String description, boolean completed) {
```
Great use overloading, I didn't think of this

hanqinilnix reviewed (commented):
Let's do this

eman-kom committed:
Pull out command classes and refactor code

eman-kom committed:
Refactor Storage class

eman-kom committed:
Fix checkstyle errors

eman-kom committed:
Made input in Ui more obvious

eman-kom committed:
Add some JUnit Tests

eman-kom committed:
Add more JUnit Tests

eman-kom committed:
Implement GUI

eman-kom committed:
Merge branch 'branch-Level-10'

eman-kom committed:
Improve code quality

More javadocs are added
Rearrange some if-else statements to make code more readable

eman-kom committed:
Add Assertions

Create assertions which are used to define assumptions about the program state so that the runtime can verify them

eman-kom committed:
Merge pull request #2 from eman-kom/branch-A-Assertions

Add Assertions

eman-kom committed:
Merge branch 'master' into branch-A-CodeQuality

eman-kom committed:
Merge pull request #1 from eman-kom/branch-A-CodeQuality

Improve code quality

eman-kom committed:
Add priority function

Able to sort after editing priority
Add priority levels to task
Add priority command to command list
Enable storage and tasklist to handle priority'

eman-kom committed:
Merge branch 'branch-A-Priority'

eman-kom committed:
Add input functionality to ListCommand to sort list by priority

eman-kom committed:
Add more checks to ensure 1 word commands are actually 1 word

eman-kom committed:
Change method name input to validate

eman-kom committed:
Fix gui

eman-kom committed:
Update GUI

eman-kom committed:
Push font to from code to fxml

eman-kom committed:
Refactor type of task to superclass

eman-kom committed:
Made GUI more readable

eman-kom committed:
Include screenshot

eman-kom committed:
Add exit function to bye

eman-kom committed:
Add Ui.png

eman-kom committed:
Add Ui.png

The previous commit did not capture the Ui.png. Hence, it is added now.

eman-kom committed:
Rename Ui.png.PNG to Ui.png

eman-kom committed:
Update README.md

eman-kom committed:
Set theme jekyll-theme-slate

eman-kom committed:
Edit Readme.md

eman-kom committed:
Merge branch 'master' of github.com:eman-kom/ip

eman-kom committed:
Fix spelling error

eman-kom committed:
Fix spelling error

eman-kom committed:
Fix spelling error

eman-kom committed:
Fix spelling error

eman-kom committed:
Fix warning from fxml

eman-kom committed:
Add Javadocs

eman-kom committed:
Fix checkstyle errors

eman-kom committed:
change font size

eman-kom committed:
file path can be now seen from gui

eman-kom committed:
Update README.md

eman-kom committed:
Update README.md

eman-kom committed:
Add acknowledgements

eman-kom committed:
Minor changes for output for task string

eman-kom committed:
Update Launcher.java
